src/github.com/matrix-org/dendrite/common/config/config.go

```diff
@@ -261,6 +252,52 @@ type ThumbnailSize struct {
 	ResizeMethod string `yaml:"method,omitempty"`
 }
 
+// Hook represents a single logrus hook. At this point, only parsing and
+// verification of the proper values for type and level are done.
+// Validity/integrity checks on the parameters are done when configuring logrus.
+type Hook struct {
+	// The type of hook, currently only "file" is supported.
+	// Yaml key is "type"
+	Type string
+
+	// The level of the logs to produce. Will output only this level and above.
+	//Yaml key is "level"
+	Level logrus.Level
+
+	// The parameters for this hook.
+	// Yaml key is "params"
+	Params map[string]interface{}
+}
+
+// UnmarshalYAML performs type coercion for a logrus hook. Additionally,
+// it ensures the type is one supported.
+func (hook *Hook) UnmarshalYAML(unmarshaler func(interface{}) error) error {
+	var tmp struct {
+		Type   string                 `yaml:"type"`
+		Level  string                 `yaml:"level"`
+		Params map[string]interface{} `yaml:"params"`
+	}
+
+	if err := unmarshaler(&tmp); err != nil {
+		return err
+	}
+
+	if tmp.Type != "file" {
```

Comment:
I wasn't really either. I am unsure about whether we sould check this in the config.go directly or in the log.go.
To get rid of the UnmarshalYAML, is that fine to just duplicate all logging elements under config.Dendrite.Derived ? (Because the log level needs processing to be retrieved)

Reply:
why not keep it as a string?